### PR TITLE
Implement env-dependent webpack configs

### PR DESCRIPTION
**Pull request by Julia-Protasevich**

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -1,38 +1,68 @@
...
+const MiniCssExtractPlugin = require('mini-css-extract-plugin');
+const webpack = require('webpack');
 
-module.exports = {
+module.exports = (env = {}) => {
+
+  const {mode = 'development'} = env;
+
+  const isProd = mode === 'production';
+  const isDev = mode === 'development';
```
Comment (MihailMartinovich) [inline webpack.config.js:10]: Move these strings to variables because you use them more than once.

File: webpack.config.js [hunk 2]
```diff
@@ -1,38 +1,68 @@
...
+
+  const getStyleLoaders = () => {
+    return [
+      isProd? MiniCssExtractPlugin.loader : 'style-loader',
```
Comment (MihailMartinovich) [inline webpack.config.js:14]: whitespace is missed. Set up code auto formatter in VS Code

File: webpack.config.js [hunk 3]
```diff
@@ -1,38 +1,68 @@
...
+        test: /\.js$/,
+        filename: '[name].js.map',
+      }));
+    }
+
+    return plugins;
+  };
+
+  return {
+    mode: isProd ? 'production' : isDev && 'development',
```
Comment (MihailMartinovich) [inline webpack.config.js:44]: instead of using ternary operator you could just pass env.mode or it has some purpose?

**Review by MihailMartinovich — CHANGES_REQUESTED** [review]

File: webpack.config.js (hunk 2 above)
Comment (Julia-Protasevich) [reply webpack.config.js]: installed and fixed

File: webpack.config.js (hunk 3 above)
Comment (Julia-Protasevich) [reply webpack.config.js]: Yup, the idea was that if someone breaks the "start" script or runs build from terminal without specifying the environment parameter, we will simply run with mode=false, so wepback will use no env-specific default optimizations

File: webpack.config.js (hunk 1 above)
Comment (Julia-Protasevich) [reply webpack.config.js]: Extracted to constants

File: webpack.config.js [hunk 4]
```diff
@@ -1,38 +1,87 @@
...
+  const DEV = 'development';
+  const PROD = 'production';
+  const NONE = 'none';
+
+  let {mode = DEV} = env;
+
+  let isProd = false;
+  let isDev = false;
+
+  const defineMode = () => {
```
Comment (MihailMartinovich) [inline webpack.config.js:16]: use pure functions everywhere

**Review by MihailMartinovich — CHANGES_REQUESTED** [review]

File: package.json [hunk 5]
```diff
@@ -4,8 +4,8 @@
   "description": "",
   "main": "index.js",
   "scripts": {
-    "start": "webpack-dev-server --open --mode development",
-    "build": "webpack --mode production"
+    "start": "webpack-dev-server --open --env.mode=development",
+    "build": "webpack --env.mod=production"
```
Comment (MihailMartinovich) [inline package.json:8]: what should be here env.mod or env.mode?

Comment (Julia-Protasevich) [reply package.json]: added 'e'

File: webpack.config.js (hunk 4 above)
Comment (Julia-Protasevich) [reply webpack.config.js]: all functions are now pure

**Pull request merged**